bungee/src/main/java/com/github/games647/fastlogin/bungee/FastLoginBungee.java

```diff
@@ -110,16 +116,22 @@ public void onDisable() {
     }
 
     private void registerHook() {
-        Plugin BungeeAuth = getProxy().getPluginManager().getPlugin("BungeeAuth");
-        if (BungeeAuth != null) {
-            core.setAuthPluginHook(new BungeeAuthHook());
-            logger.info("Hooked into BungeeAuth");
-        }
-        Plugin BungeeCordAuthenticatorBungee = getProxy().getPluginManager().getPlugin("BungeeCordAuthenticatorBungee");
-        if (BungeeCordAuthenticatorBungee != null) {
-            logger.info("Try to hook into BungeeCordAuthenticatorBungee...");
-            BungeeCordAuthenticatorHook hook = new BungeeCordAuthenticatorHook(BungeeCordAuthenticatorBungee, logger);
-            core.setAuthPluginHook(hook);
+        try {
+            List<Class<? extends AuthPlugin<ProxiedPlayer>>> hooks = Arrays.asList(
+                    BungeeAuthHook.class, BungeeCordAuthenticatorBungeeHook.class, SodionAuthHook.class);
+
+            for (Class<? extends AuthPlugin<ProxiedPlayer>> clazz : hooks) {
+                String pluginName = clazz.getSimpleName();
+                pluginName = pluginName.substring(0, pluginName.length() - 4);
+                //uses only member classes which uses AuthPlugin interface (skip interfaces)
+                Plugin plugin = getProxy().getPluginManager().getPlugin(pluginName);
+                if (plugin != null) {
+                    core.setAuthPluginHook(
```

Comment:
A logging statement would be useful to inform the users about the plugin name, we hooked into.